src/Features/CSharp/Portable/Copilot/CSharpCopilotNotImplementedMethodFixProvider.Analyzer.cs

```diff
@@ -0,0 +1,157 @@
+                    if (refRoot == null)
+                        return null;
+
+                    var referenceNode = refRoot.FindNode(l.Location.SourceSpan);
+                    var containingMethod = referenceNode.Ancestors().OfType<MethodDeclarationSyntax>().FirstOrDefault();
+                    var refText = await refDocument.GetTextAsync(cancellationToken).ConfigureAwait(false);
+
+                    var contextSpan = GetContextSpan(refText, l.Location.SourceSpan, containingMethod);
+                    return new MethodImplementationReferenceContext
+                    {
+                        FileName = refDocument.Name,
+                        SurroundingCode = refText.ToString(contextSpan)
+                    };
+                })
+                .Where(x => x != null)
+                .ToList();
+
+            var referenceContexts = await Task.WhenAll(topReferences).ConfigureAwait(false);
+
+            // Get previous and next tokens for context
+            var previousToken = methodDeclaration.Modifiers.Count > 0
+                ? methodDeclaration.Modifiers[0]
+                : methodDeclaration.GetFirstToken();
+            var nextToken = methodDeclaration.GetLastToken();
+
+            // Extract method information
+            var parameters = methodDeclaration.ParameterList.Parameters;
+
+            // Get C# language version
+            var parseOptions = (CSharpParseOptions?)document.Project.ParseOptions;
+            var languageVersion = parseOptions?.LanguageVersion.ToString() ?? string.Empty;
+
+            // Create analysis record
+            var record = new MethodImplementationProposal
+            {
+                MethodName = methodDeclaration.Identifier.Text,
+                MethodBody = methodDeclaration.Body?.ToString() ?? string.Empty,
+                ReturnType = methodDeclaration.ReturnType.ToString(),
+                Parameters = parameters.Select(p => new MethodImplementationParameterContext
+                {
+                    Name = p.Identifier.Text,
+                    Type = p.Type?.ToString() ?? string.Empty,
+                    Modifiers = p.Modifiers.Select(m => m.Text).ToImmutableArray()
+                }).ToImmutableArray(),
+                ReferenceCount = referenceCount,
+                TopReferences = referenceContexts?.Where(x => x != null).Select(x => x!).ToImmutableArray() ?? [],
+                ContainingType = methodDeclaration.Parent?.ToString() ?? string.Empty,
+                Accessibility = GetAccessibility(methodDeclaration.Modifiers).ToString().ToLower(),
+                Modifiers = methodDeclaration.Modifiers.Select(m => m.Text).ToImmutableArray(),
+                PreviousTokenText = previousToken.Text,
+                NextTokenText = nextToken.Text,
+                LanguageVersion = languageVersion,
+            };
+
+            return record;
+        }
```

Comment:
skipped to here, as dustin gave a lot of feedback on this.  i can review htis again after your next pass.